### ISSUE TITLE
Match aria-selected to what is programmatically selected for Location, Community Provider > Service type, and VA Health > Service type fields

### ISSUE THREAD

**Issue by qjac** [department-of-veterans-affairs/va.gov-cms#23483]
## Describe the defect
Currently aria-selected is updating to true as the user navigates through the dropdown list, but programmatic selection happens on click. 

Based on [research](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/22893), and in line with VADS, we want to set aria-selected to true only when that value is also programmatically selected.

## To Reproduce
1. Go to /find-locations
2. Open devtools and explore the HTML of a custom combobox (Location, Community Provider > Service type, and VA Health > Service type fields)
3. Type into the field or click to open dropdown
4. Navigate through the dropdown with arrow keys or mouse
5. Note that the aria-activedescendant value set on the input matches the id of the currently highlighted dropdown option
6. Note that the aria-selected value on the currently highlighted item updates to true when highlighted
7. Select an option and reopen dropdown
8. Note that upon first reopening dropdown, no options have aria-selected set to true even though an option is selected
9. Navigate through the dropdown
10. Note that aria-selected is once again updating to match aria-activedescendant ad not the actually selected value

## AC / Expected behavior
- [ ] Location
  - [ ] aria-selected is set to false until a value is programmatically selected. 
  - [ ] Once a selection is made, then aria-selected updates to true
- [ ] Community Provider > Service type
  - [ ] aria-selected is set to false until a value is programmatically selected. 
  - [ ] Once a selection is made, then aria-selected updates to true
- [ ] VA Health > Service type
  - [ ] aria-selected is set to false until a value is programmatically selected. 
  - [ ] Once a selection is made, then aria-selected updates to true

## Screenshots
https://github.com/user-attachments/assets/840f4c12-1a2e-426a-9956-e988b7ecb28d

 ## Additional context
[Related slack convo 🔒](https://dsva.slack.com/archives/C0FQSS30V/p1770760296183129)

**Comment by carlyMhayter** [comment]
This issue was affected by [[FL] Standardize the number of characters typed that open the dropdowns initially](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/22889)...because for step 7, the dropdown does not re-open when it's clicked or tabbed into. You have to type (or change the value of the input) in order to trigger the dropdown...I'll ask about the UX regarding whether we want the dropdown to open once we click into the input and an option is already selected

**Comment by qjac** [comment]
I don't think it matters. I've seen both in the wild. It's only relevant when the screen reader is traversing the dropdown, so when it's open, but shouldn't hurt anything if it persists when it closes.
